### PR TITLE
Feature/timeline controls

### LINKED ISSUE #31: Filters and Search for the Timeline
It should be possible to filter the events in the timeline component by typing in some search keywords or by setting defined filters.

### LINKED ISSUE #32: Download Event Data of the Timeline
The timeline component should provide a button that enables the user to download the events as CSV.

### PR DESCRIPTION
**Pull request by chaoran-chen**
Here is the next version of the timeline component.

* It contains a field to filter events (closes #31).
* It is possible to download the filtered events as CSV (closes #32).
* There is now a full-screen mode.

timeline

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+language: node_js
+script: echo "It's alright!"
```
Comment (chaoran-chen) [inline .travis.yml:2]: Can someone explain why Github / Git shows these lines as new and does not recognize it as already given in the current state of the develop-branch? And does it mean that I should not use rebase to get a feature-branch "updated"?

Comment (repus) [inline .travis.yml]: Published I think.

**Review by lustoykov — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (lustoykov) [reply .travis.yml:2]: I can't answer your first question (sure you rebased correctly?), but about your second

> And does it mean that I should not use rebase to get a feature-branch "updated"

No, in fact rebase on top of master / develop is perfect to get a feature-branch updated.

Comment (martomi) [reply .travis.yml:2]: I think you might have rebased your changes on another feature branch instead of develop for this to happen?

Comment (chaoran-chen) [reply .travis.yml:2]: Oh, yes, that's possible. Thanks!

**Pull request merged**